### ISSUE TITLE
Dependency Dashboard reporting merged items as closed/unmerged

### ISSUE THREAD

**Issue by adam-moss** [renovatebot/renovate#3365]
**What Renovate type are you using?**
Renovate cli via the official docker image

**Describe the bug**
I've added the masterIssue and masterIssueTitle attributes to my config.js. When reviewing the master issue created it has list 3 MRs as closed without being merged, however when you click through to the status of these it shows they have actually been merged.

**Did you see anything helpful in debug logs?**
Sadly not, but I can do some more digging.

**Expected behavior**
Summary would provide accurate results.

**Additional context**
This is running against gitlab on-prem.

**Comment by rarkins** [comment]
You should only get entries in masterIssue if Renovate thinks there are pending updates. If you're seeing them in masterIssue but you say the linked MRs were already merged, it implies that you are missing out on updates because Renovate thinks you already merged them previously. Is that the case, or is there some other explanation for the situation?

**Comment by adam-moss** [comment]
Not sure, nothing special I can see other than they were merged by a human rather than renovate itself 🤔

**Comment by rarkins** [comment]
If they were merged, then why would there be new PRs waiting to update to the exact same version? i.e. did you roll them back, did you add new package files with the older versions, etc?

**Comment by adam-moss** [comment]
That's the point, there isn't new MR's, there is just items in the master issue that suggests there should be, but link to already-done merges 👍

**Comment by rarkins** [comment]
Can you debug it like this?

1. Click through and look at the branch name being used for one of these MRs
2. Look through the debug logs for any reference to that branch name

**Comment by rarkins** [comment]
Could you also screenshot what you see? Redact the exact MR titles if necessary of course.

**Comment by adam-moss** [comment]
Couple of screen caps:

**Master Issue**
![Screen Shot 2019-03-15 at 14 12 25](https://user-images.githubusercontent.com/2951486/54437433-8494fb00-472c-11e9-90c5-a83e9c87b58a.png)

**First MR Referenced**
![Screen Shot 2019-03-15 at 14 12 39](https://user-images.githubusercontent.com/2951486/54437442-8959af00-472c-11e9-9e3c-bd303b14e2d4.png)

Working on getting some debug logs also 👍

**Comment by rarkins** [comment]
What I think is happening is that there actually *are* new MRs ready, e.g. to bump the digests. Have you customised the commitMessage or prTitle fields?

**Comment by adam-moss** [comment]
Commit message is customised, yes:

```
    "commitMessage": "{{commitMessagePrefix}} update to {{newValue}}",
    "docker": {
      "pin": {
        "commitMessage": "{{commitMessagePrefix}} pin image digest",
      }
    },
    "pin": {
      "commitMessage": "{{commitMessagePrefix}} pin dependency version",
    }
```

Nothing outstanding from an MR point of view:
![Screen Shot 2019-03-15 at 14 40 13](https://user-images.githubusercontent.com/2951486/54439245-56b1b580-4730-11e9-8403-0af425d30bed.png)

**Comment by rarkins** [comment]
I think the new ones are being blocked, and it's because of the commitMessage customisation. With the template you have, digest updates are not uniquely identified and therefore get blocked by previously open ones. i.e if the old MR is named "update to dind" and the new one also would have MR title "update to dind" then it's considered blocked.

The master issue itself though is a little confused by this blocking. We could cosmetically fix it, but ultimately you are still missing out on digest updates because of the custom commit messages.

If you were to tick one of those MRs, then I think you'd find a MR created and indeed the digest would be (validly) changed.

If you were to remove the custom commit messages then I think you'd find all three opened and including the digest value in the MR title.

**Comment by adam-moss** [comment]
Yeah, I think you're right, I'm going to do some more testing on this.  Should be fun 😈

**Comment by rarkins** [comment]
In the early days you had no choice but to replace the entire `commitMessage` if you wanted it changed, but a while ago I broke it into sub-templates to make it easier to change things without breaking the "has this PR been opened before?" checks, etc. I think if you revert back to default you can then find simpler ways to customise if you like, including per-manager.

**Comment by adam-moss** [comment]
Yup, I've removed them all of now and will re-review if we actually need to change them, I don't think we do anymore.

I wonder if it is worth building in a check for when this circumstance occurs and raising an error?  In any event this ticket can be closed now I think.

**Comment by rarkins** [comment]
Renovate should already warn that directly editing commitMessage is deprecated, so I don't think there's more it can do there.

But in this case it incorrectly reported that the other PRs were *closed* instead of *merged* so I think that "cosmetic" problem should be fixed so we can leave this open.

**Comment by adam-moss** [comment]
Yeah I can see that in the docs, but can't see that in the renovate output as it runs.  But what I meant is this has came to light because of a cosmetic issue in the master issue, but if I hadn't enabled that I'd be blissfully unaware it was happening.  So should a warning be raised when Renovate determines it is trying to up-date something that is already merged, rather than the no-op as when something is closed?

**Comment by rarkins** [comment]
Renovate logs whenever a MR is blocked, you should be able to notice that if you are checking into level logs. I have considered whether to post a comment to the merged PR informing that it’s blocking something like we do with closed ones. Apart from that not much we can do I think

**Comment by rarkins** [comment]
Dashboard status/categorization problems should be resolved as of renovate@25.37.1
